lib/tapioca/dsl/compilers/active_record_typed_store.rb

```diff
@@ -95,11 +95,20 @@ class ActiveRecordTypedStore < Compiler
         sig { override.void }
         def decorate
           stores = constant.typed_stores
-          return if stores.values.flat_map(&:accessors).empty?
+          accessors = stores.values.flat_map(&:accessors)
+          return if accessors.empty?
+
+          # Support versions > v1.5.0
+          if accessors.first.is_a?(Hash)
+            return if accessors.filter_map { |accessor| accessor.keys.first }.empty?
+          end
 
           root.create_path(constant) do |model|
             stores.values.each do |store_data|
               store_data.accessors.each do |accessor|
+                # Support versions > v1.5.0
+                accessor = accessor.first if accessor.is_a?(Array)
```

Comment:
This won't work if the accessors use a prefix or suffix. The method name generated for the shim will be based on the accessor, not on the actual method (which is the value in the Hash, so that last element of the two-element array `accessor` here).